src/main/scala/loamstream/drm/lsf/BacctAccountingClient.scala

```diff
@@ -180,6 +188,32 @@ object BacctAccountingClient {
     val startTime = "(.*):.*[Dd]ispatched\\sto.*".r
     //Thu Apr 18 22:32:01: Completed <exit>.
     val endTime = "(.*):\\s+Completed.*".r
+    //Mon May  6 22:58:03: Completed <exit>; TERM_RUNLIMIT: job killed after reaching LSF run time limit.
+    val termReason = ".*:\\s+Completed.*;\\s+(TERM_.*):.*".r
+    //MEM SWAP
+    //3M  60M
     val memory = "^(.+)[MmGg]$".r
   }
+  
+  //See https://www.ibm.com/support/knowledgecenter/en/SSWRJV_10.1.0/lsf_command_ref/bacct.1.html
+  private[lsf] def parseTerminationReason(lsfReason: String): TerminationReason = lsfReason.trim.toUpperCase match {
+    //Job was killed after it reached LSF CPU usage limit (12)
+    case raw @ "TERM_CPULIMIT" => TerminationReason.CpuTime(Option(raw))
```

Comment:
Just curious why these take an `Option[String]`, and why that string is `raw`? I'm probably missing some context, but it feels like there's data being lost and/or duplicated?

```scala
case raw @ "FOO" => Foo(Option(raw))
```

It feels like what you want this to be is:

```scala
def toTerminationReason(bacctOutput: Seq[String]): Option[TerminationReason] = {
  bacctOutput
    .collectFirst { case fullString @ Regexes.termReason(r) => fullString -> r }
    .map { case (fullString, r) => parseTerminationReason(fullString, r) }
}

def parseTerminationReason(fullString: String, lsfReason: String): TerminationReason = 
  lsfReason.trim.toUpperCase match {
    case "TERM_CPULIMIT" => TerminationReason.CpuTime(fullString)
    // ...
  }
}
```

It feels like you want the entire error string as the member of the termination reason, and the matched "reason" as the ADT value. And that you don't want the full reason string to be optional (when would it be `None`?).

Again, that's just a guess, though.

Reply:
Just noticing the parameter name is `lsfReason` and that might be why you have `Option` (because Uger and Google may not have text to go there)? Still, I think my original comment stands?

Reply:
I'm glad you noticed this, as it always felt funny to me too.  
> Just noticing the parameter name is lsfReason and that might be why you have Option (because Uger and Google may not have text to go there)? 
That's basically it.  I wanted to keep whatever value from the underlying environment was used to decide on a particular ADT value.  That might be a string, or it might not be anything at all -  one API for accessing Uger just gives you a bunch of flags, like `isTerminated`, etc, and doesn't say _why_.  LS will have to use heuristics to guess a reason in those cases.  

That said, it's not clear that those underlying values are useful.  The LSF codes `TERM_MEMORY` and `TERM_SWAP` are both considered "memory" by LS, but does anyone care?

The issue here to me feels like while `TerminationReason` is a cross-environment concept, so LS can have common logic for deciding whether to re-run a failed job based on why it failed, but the actual ADT _values_ are environment-specific.  Maybe the thing to do is something like:
```scala
sealed trait TerminationReason

sealed abstract class LsfTerminationReason(raw: String) extends TerminationReason

final case class LsfMemory(raw: String) extends LsfTerminationReason(raw)
...

sealed trait UgerTerminationReason extends TerminationReason
...
sealed trait GoogleTerminationReason extends TerminationReason
```
But then exhaustiveness checks are out the window.  It feels like what I wanted was some abstract, cross-environment contract for the "shape" of the ADT (allowed values, etc) and environment-specific _implementations_ of the ADT.  I bet that's possible, but I worry it might get gnarly.  

I'll see what I can come up with here.  Thanks for the suggestions!

Reply:
> That said, it's not clear that those underlying values are useful.

I do think that the full error text from the output could be quite useful (date/times, added reason text beyond the "type", etc.) to Ryan or anyone debugging. But, no, I would agree that I don't think the raw "type" text is useful to anyone if the ADT supersedes it.

Reply:
I'll get rid of the raw stuff, since extra stuff 
> date/times, added reason text beyond the "type", etc.
is collected for all jobs after the next PR.  There's no point collecting the raw field here if Ryan will never look at it and will instead consult the `qacct` output (or whatever) which contains much more info.